tests/rptest/services/kgo_verifier_services.py

```diff
@@ -0,0 +1,522 @@
+
+import os
+import json
+import threading
+import requests
+from enum import Enum
+
+from rptest.services.redpanda import RedpandaService
+from ducktape.services.service import Service
+from ducktape.utils.util import wait_until
+from ducktape.cluster.remoteaccount import RemoteCommandError
+
+# Install location, specified by Dockerfile or AMI
+TESTS_DIR = os.path.join("/opt", "kgo-verifier")
+
+# TODO: need to update terraform to open a range of ports
+REMOTE_PORT_BASE = 8080
+
+
+class KgoVerifierService(Service):
+    """
+    KgoVerifierService is kgo-verifier service.
+    To validate produced record user should run consumer and producer in one node.
+    Use ctx.cluster.alloc(ClusterSpec.simple_linux(1)) to allocate node and pass it to constructor
+    """
+    def __init__(self, context, redpanda, topic, msg_size, custom_node):
+        self.use_custom_node = custom_node is not None
+
+        # We should pass num_nodes to allocate for our service in BackgroundThreadService,
+        # but if user allocate node by themself, BackgroundThreadService should not allocate any nodes
+        nodes_for_allocate = 1
+        if self.use_custom_node:
+            nodes_for_allocate = 0
+
+        super(KgoVerifierService, self).__init__(context,
+                                                 num_nodes=nodes_for_allocate)
+
+        # Should check that BackgroundThreadService did not allocate anything
+        # and store allocated nodes by user to self.nodes
+        if self.use_custom_node:
+            assert not self.nodes
+            self.nodes = custom_node
+
+        self._redpanda = redpanda
+        self._topic = topic
+        self._msg_size = msg_size
+        self._pid = None
+        self._remote_port = None
+
+        for node in self.nodes:
+            if not hasattr(node, "kgo_verifier_next_port"):
+                node.kgo_verifier_next_port = REMOTE_PORT_BASE
+
+    @property
+    def log_path(self):
+        return f"/tmp/{self.who_am_i()}.log"
+
+    @property
+    def logs(self):
+        return {
+            "kgo_verifier_output": {
+                "path": self.log_path,
+                "collect_default": True
+            }
+        }
+
+    def spawn(self, cmd, node):
+        assert self._pid is None
+
+        port = node.kgo_verifier_next_port
+        node.kgo_verifier_next_port += 1
+        self._remote_port = port
+
+        wrapped_cmd = f"nohup {cmd} --remote --remote-port {port} > {self.log_path} 2>&1 & echo $!"
+        self.logger.debug(f"spawn {self.who_am_i()}: {wrapped_cmd}")
+        pid_str = node.account.ssh_output(wrapped_cmd)
+        self.logger.debug(f"spawned {self.who_am_i()} pid={pid_str}")
+        pid = int(pid_str.strip())
+        self._pid = pid
+
+    def stop_node(self, node, **kwargs):
+        if self._status_thread:
+            self._status_thread.stop()
+            self._status_thread = None
+
+        if self._pid is None:
+            return
```

Comment:
shouldn't it throw an exception?

Reply:
This is a legal path: if for example a service was instantiated but never started (perhaps we had a test exception before that happened), then ducktape will still call stop on it during teardown.